### ISSUE TITLE
Add a function for us to import our wallet back

### ISSUE THREAD

**Issue by gillbates** [MacWallet/MacWallet#11]
Since there's already an export function, can we also have an import as well? 
Thanks in advance!

**Comment by paolovacirca** [comment]
My wallet is corrupt. I can't unlock it, although it says it's already unlocked. And I can't lock it either. Nor can I transfer money, because it says "wrong password" without even letting me type in any password at all. And my Bitcoin is stuck in there, I can't get it back.

Please, let me import my backed up wallet somehow!

**Comment by gillbates** [comment]
Oaiki, we have the same situation here..so please provide an import function somehow!

**Comment by jonasschnelli** [comment]
I'm working on it and will be release a version with import functions soon

**Comment by gillbates** [comment]
Jonas, up to today, there are 18 friends of me (They followed my suggestions to switch to MacWallet 3 months ago) and I are experiencing the same problem as MrOaiki reported. There are about 1600 BTC stuck there ... Please roll it out asap...
Thanks in advance!

**Comment by jonasschnelli** [comment]
@gillbates: did you made additional addresses? Did you made/have a backup when you wallet or additional address was created?

**Comment by jonasschnelli** [comment]
If the private key was getting corrupt, a newer version won't help. The only hope is then that you have a backup.

**Comment by gillbates** [comment]
we've exported the keys using macwallet.
And the file looks like:

pub: 0xxxxxx
priv: 0xxxxx

are these info enough?

**Comment by jonasschnelli** [comment]
Yes. You can also import this keys into bitcoind or multibit.

Read: https://en.bitcoin.it/wiki/How_to_import_private_keys
Or: https://multibit.org/help_importASingleKey.html

**Comment by paolovacirca** [comment]
@jonasschnelli How do I know if the private key is corrupt or not? I have several backups, with both pub: xxxx and priv: 0000, but the application is stuck (as described above). If there is a risk that the backups are corrupt as well, then how would a backup help to begin with?

And no, you can NOT import your keys into Multibit. I have tried. I guess it's because they are encrypted with MacWallet.

**Comment by gillbates** [comment]
@jonasschnelli I followed https://multibit.org/help_importASingleKey.html and the multibit says
the private keys unlock failed. The error was "Could not understand address in import file"

**Comment by jonasschnelli** [comment]
@gillbates did you also try with bitcoind?

**Comment by gillbates** [comment]
@jonasschnelli yes. But it shows ￼ Invalid private key (code -5) in the console.
I've compared the keys exported by multibit and by MacWallet.
It seemed MacWallet's key has 12 letters more ...

**Comment by jonasschnelli** [comment]
Okay... i'll will investigate in that.
As long as there is a private key a import will be possible somehow.
Will give you news soon.

**Comment by gillbates** [comment]
@jonasschnelli please hurry ...

**Comment by paolovacirca** [comment]
@jonasschnelli Just one last question… I have several wallet.txt exports, and there are various keys. The one where my bitcoins are located in, however, seems to simply not exist in the backup. It's as if it never exported the private key. Does this mean it's nowhere to be found on my computer? It's forever gone?

**Comment by jonasschnelli** [comment]
It's very hard to say from my point of view.
I just hope you did use MacWallt (with version around 0.1) with extremely care.

An export is not a backup.
Did you never back up the wallet file in you ~/Library?
Do you use system wide backups like timemachine?

When you run MacWallet, is the address where you bitcoind are related to visible in the address list?

If yes, did you made a export or backup after right creating this address?

**Comment by paolovacirca** [comment]
@jonasschnelli No, I never backed up my wallet file in ~/Library, I thought exporting it was backup enough. I'm not using any system wide backups, I use Dropbox for backing up important folders (of which MacWallets library was not one, which I now realize was a mistake).

"When you run MacWallet, is the address where you bitcoind are related to visible in the address list?"
Yes, it is visible in the address list.

 "If yes, did you made a export or backup after right creating this address?"
Yes, I did, however… When I now look in the exported file, there are pub and priv keys for all wallets except the one where my bitcoin is. That wallet only has a pub key in the wallet.txt file.
